lib/mongoose.providers.ts

```diff
@@ -1,20 +1,7 @@
 import { Provider } from '@nestjs/common';
 import { Connection, Document, Model } from 'mongoose';
 import { getConnectionToken, getModelToken } from './common/mongoose.utils';
-import {
-  AsyncModelFactory,
-  DiscriminatorOptions,
-  ModelDefinition,
-} from './interfaces';
-
-function addDiscriminators(
```

Comment:
could probably keep this function, just change it to: 
```
function addDiscriminators(
  discriminators: DiscriminatorOptions[] = [],
  option: ModelDefinition | AsyncModelFactory,
) {
  return discriminators.map((d) => ({
    provide: getModelToken(d.name),
    useFactory: (model: Model<Document>) =>
      model.discriminator(d.name, d.schema),
    inject: [getModelToken(option.name)],
  }));
}
```
and then just spread it's return `...addDiscriminators(option.discriminators, option)`